tests/cpp/StreamTests.cpp

```diff
@@ -142,7 +142,23 @@ BOOST_AUTO_TEST_CASE(testErrorOnUnsupportedUncompressedFormats)
     {
         deflect::ImageWrapper image(pixels.data(), 4, 4, format);
         image.compressionPolicy = deflect::COMPRESSION_OFF;
-        BOOST_CHECK(!stream.send(image).get());
+        BOOST_CHECK_THROW(stream.send(image).get(), std::invalid_argument);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(testSuccessOnCompressedFormats)
+{
+    deflect::Stream stream("id", "dummyhost");
+    std::vector<unsigned char> pixels(4 * 4 * 4);
+
+    std::vector<deflect::PixelFormat> unsupportedUncompressedFormats = {
```

Comment:
name it allFormats here? I think you can even use auto for the variable type (init list).